projects/auth0-angular/src/lib/auth.service.ts

```diff
@@ -43,42 +44,52 @@ import { AuthClientConfig } from './auth.config';
 })
 export class AuthService implements OnDestroy {
   private isLoadingSubject$ = new BehaviorSubject(true);
-  private isAuthenticatedSubject$ = new BehaviorSubject(false);
   private errorSubject$ = new ReplaySubject<Error>(1);
+  private refreshState$ = new Subject();
 
   // https://stackoverflow.com/a/41177163
   private ngUnsubscribe$ = new Subject();
-
   /**
    * Emits boolean values indicating the loading state of the SDK.
    */
   readonly isLoading$ = this.isLoadingSubject$.asObservable();
 
   /**
-   * Emits boolean values indicating the authentication state of the user. If `true`, it means a user has authenticated.
-   * This depends on the value of `isLoading$`, so there is no need to manually check the loading state of the SDK.
+   * Trigger used to pull User information from the Auth0Client.
+   * Triggers when an event occurs that needs to retrigger the User Profile information.
+   * Such as: Initially, getAccessTokenSilently, getAccessTokenWithPopup and Logout
    */
-  readonly isAuthenticated$ = this.isLoading$.pipe(
+  private isAuthenticatedTrigger$ = this.isLoading$.pipe(
     filter((loading) => !loading),
     distinctUntilChanged(),
-    concatMap(() => this.isAuthenticatedSubject$)
+    concatMap(() =>
+      this.refreshState$.pipe(
+        mergeMap(() => this.auth0Client.isAuthenticated())
+      )
+    )
+  );
+
+  /**
+   * Emits boolean values indicating the authentication state of the user. If `true`, it means a user has authenticated.
+   * This depends on the value of `isLoading$`, so there is no need to manually check the loading state of the SDK.
+   */
+  readonly isAuthenticated$ = this.isAuthenticatedTrigger$.pipe(
+    distinctUntilChanged()
```

Comment:
The one in `isAuthenticatedTrigger$` is used to block non-distinct subsequent values from `isLoading$` (which shouldnt happen anyway, but I dont think its bad to keep it).

Considering the isAuthenticatedTrigger$ is built from this.refreshState$, it is expected that it will still emit non-distinct subsequent values. This is because we need this to re-emit the user and idTokenClaims observable in case `isAuthenticatedTrigger` emits `true` followed by another `true`.

However, we want our `isAuthenticated` observable to still only emit distinct values.